File: Python/solvedac.py
class Solvedac:
    levels = ['Unrated']
    level = ['브론즈', '실버', '골드', '플래티넘', '다이아몬드', '루비']

    def __init__(self) -> None:
        self.url = "https://solved.ac/api/v3/problem/show"
        self.headers = {"Content-Type": "application/json"}

        for Lv in self.level:
            for idx in range(5, 0, -1):
                self.levels.append(Lv+str(idx))

File: Plugin/scripts/SolvedAC.js
export default class SolvedAC {
  constructor() {
    this.SOLVEDAC_BASE_URL = "https://solved.ac/api/v3";
  }

  async getProblemInfo(problemId) {
    validate(
      problemId,
      "Invalid problem ID for requesting problem information."
    );

    const url = `${this.SOLVEDAC_BASE_URL}/problem/show?problemId=${problemId}`;
    const headers = {
      Accept: "application/json",
    };

    const response = await request(url, "GET", headers, undefined);
    const data = await response.json();
    const problemInfo = {
      problemId: data.problemId,
      level: data.level,
      title: data.titleKo,
      tags: data.tags,
    };

    return problemInfo;
  }
}

async function request(url, method, headers, body) {
  return await fetch(url, {
    method: method,
    headers: headers,
    body: body,
  });
}

function validate(value, msg) {
  if (isEmpty(value)) {
    throw new Error(`[SolvedAC] : ${msg}`);
  }
}

function isEmpty(value) {
  return (
    !value || (typeof value === "object" && Object.keys(value).length === 0)
  );
}
